src/main/java/com/coffee/coffeeserviceproject/member/service/MemberService.java

```diff
@@ -0,0 +1,144 @@
+package com.coffee.coffeeserviceproject.member.service;
+
+import static com.coffee.coffeeserviceproject.common.type.ErrorCode.ALREADY_EXISTS_USER;
+import static com.coffee.coffeeserviceproject.common.type.ErrorCode.LOGIN_ERROR;
+import static com.coffee.coffeeserviceproject.common.type.ErrorCode.WRONG_PASSWORD;
+import static com.coffee.coffeeserviceproject.member.type.RoleType.BUYER;
+import static com.coffee.coffeeserviceproject.member.type.RoleType.SELLER;
+
+import com.coffee.coffeeserviceproject.common.exception.CustomException;
+import com.coffee.coffeeserviceproject.member.dto.MemberDeleteDto;
+import com.coffee.coffeeserviceproject.member.dto.MemberDto;
+import com.coffee.coffeeserviceproject.member.dto.MemberUpdateDto;
+import com.coffee.coffeeserviceproject.member.dto.RoasterDto;
+import com.coffee.coffeeserviceproject.member.entity.Member;
+import com.coffee.coffeeserviceproject.member.entity.Roaster;
+import com.coffee.coffeeserviceproject.member.repository.MemberRepository;
+import com.coffee.coffeeserviceproject.util.JwtUtil;
+import com.coffee.coffeeserviceproject.util.PasswordUtil;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+
+  private final MemberRepository memberRepository;
+
+  private final MailService mailService;
+
+  private final JwtUtil jwtUtil;
+
+  public void addMember(MemberDto memberDto) {
+
+    Optional<Member> optionalMember = memberRepository.findByEmail(memberDto.getEmail());
+    if (optionalMember.isPresent()) {
+      throw new CustomException(ALREADY_EXISTS_USER);
+    }
```

Comment:
이런형태는 지양하면 좋습니다. Optional로 선언하는 변수는 없는 것이 좋은 코딩입니다.
```suggestion
    if (memberRepository.findByEmail(memberDto.getEmail())) {
      throw new CustomException(ALREADY_EXISTS_USER);
    }
```

Reply:
> 이런형태는 지양하면 좋습니다. Optional로 선언하는 변수는 없는 것이 좋은 코딩입니다.

앗.. 감사합니다 리팩토링 하면서 빼먹었네요.. 바로 수정하겠습니다